### PR TITLE
Add test templates for DRA-enabled clusters

### PR DESCRIPTION
**Pull request by nojnhuh**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

This PR adds new test templates that describe clusters with DRA enabled for CI versions and custom builds of Kubernetes. A forthcoming test-infra PR will create a new periodic job to run the k/k DRA e2e tests using these templates.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/conformance_test.go [hunk 1]
```diff
@@ -97,21 +97,22 @@ var _ = Describe("Conformance Tests", func() {
 			Expect(err).NotTo(HaveOccurred())
 			Expect(os.Setenv("CI_VERSION", kubernetesVersion)).To(Succeed())
 			Expect(os.Setenv("CLOUD_PROVIDER_AZURE_LABEL", "azure-ci")).To(Succeed())
+		}
 
+		if flavor == "" {
```
Comment (jackfrancis) [inline test/e2e/conformance_test.go:102]: are these changes related to the new templates?

Comment (nojnhuh) [reply test/e2e/conformance_test.go:102]: Yes, without this then any run passing the `kubetest.use-ci-artifacts` or `kubetest.use-pr-artifacts` flags to the e2e binary will only use the vanilla ci-version/custom-builds template.

**Comment by nojnhuh** [comment]
tests are green!

/hold cancel
